pkg/kubelet/status/status_manager.go

```diff
@@ -542,11 +560,15 @@ func checkContainerStateTransition(oldStatuses, newStatuses []v1.ContainerStatus
 // updateStatusInternal updates the internal status cache, and queues an update to the api server if
 // necessary. Returns whether an update was triggered.
 // This method IS NOT THREAD SAFE and must be called from a locked function.
-func (m *manager) updateStatusInternal(pod *v1.Pod, status v1.PodStatus, forceUpdate bool) bool {
+func (m *manager) updateStatusInternal(pod *v1.Pod, status v1.PodStatus, forceUpdate, podIsFinished bool) bool {
 	var oldStatus v1.PodStatus
 	cachedStatus, isCached := m.podStatuses[pod.UID]
 	if isCached {
 		oldStatus = cachedStatus.status
+		// once terminated pod continues to be terminated
```

Comment:
is there a reason you needed this?  The only caller of updateStatusInternal should be sync loop or use the last cached status.

I think this is a panic condition / should be impossible.  Can you make this a panic while we test the PR to see when this case is happening (cached is terminated, new status isn't), and then once we get it verified we should turn this into a klog.ErrorS("programmer error ...")

Reply:
Yes, I'm going to check that. For now I will just answer that I was thinking about a scenario when syncBatch triggers SyncPod just after SyncTerminatedPod.

Reply:
Added a panic and it seems all tests pass. I will replace it with the error message before merge.